client/src/modules/account_statement/account_statement.ctrl.js

```diff
@@ -0,0 +1,288 @@
+      visible          : false },
+
+    { field            : 'period_end',
+      displayName      : 'TABLE.COLUMNS.PERIOD',
+      headerCellFilter : 'translate',
+      cellTemplate     : 'modules/templates/bhPeriod.tmpl.html',
+      visible          : false },
+
+    { field                : 'hrRecord',
+      displayName          : 'TABLE.COLUMNS.RECORD',
+      headerCellFilter     : 'translate',
+      visible              : false,
+      footerCellTemplate   : '<i></i>' },
+
+    { field            : 'currencyName',
+      displayName      : 'TABLE.COLUMNS.CURRENCY',
+      headerCellFilter : 'translate',
+      visible          : false },
+
+    { field            : 'debit',
+      displayName      : 'TABLE.COLUMNS.DEBIT_SOURCE',
+      headerCellFilter : 'translate',
+      visible          : false,
+      cellTemplate     : '/modules/journal/templates/debit.grid.html' },
+
+    { field            : 'credit',
+      displayName      : 'TABLE.COLUMNS.CREDIT_SOURCE',
+      headerCellFilter : 'translate',
+      visible          : false,
+      cellTemplate     : '/modules/journal/templates/credit.grid.html' },
+
+    { field                : 'hrEntity',
+      displayName          : 'TABLE.COLUMNS.RECIPIENT',
+      headerCellFilter     : 'translate',
+      editableCellTemplate : '/modules/journal/templates/entity.edit.html',
+      visible              : true },
+
+    { field            : 'hrReference',
+      displayName      : 'TABLE.COLUMNS.REFERENCE',
+      headerCellFilter : 'translate',
+      visible          : true },
+
+    { field            : 'origin_id',
+      displayName      : 'FORM.LABELS.TRANSACTION_TYPE',
+      headerCellFilter : 'translate',
+      cellTemplate     : '/modules/journal/templates/transaction_type.html',
+      visible          : false },
+
+    { field            : 'comment',
+      displayName      : 'FORM.LABELS.COMMENT',
+      headerCellFilter : 'translate',
+      visible          : true },
+
+    { field            : 'display_name',
+      displayName      : 'TABLE.COLUMNS.RESPONSIBLE',
+      headerCellFilter : 'translate',
+      visible          : false },
+  ];
+
+  vm.gridOptions.columnDefs = columns;
+
+  // on register api
+  function onRegisterApi(api) {
+    vm.gridApi = api;
+    vm.gridApi.selection.on.rowSelectionChanged(null, rowSelectionChanged);
+    vm.gridApi.selection.on.rowSelectionChangedBatch(null, rowSelectionChanged);
+  }
+
+  // row selection changed
+  function rowSelectionChanged() {
+    vm.selectedRows = vm.gridApi.selection.getSelectedGridRows();
+  }
+  // end grid defintion =============================================================
+
+  // comment selected rows
+  vm.commentRows = function commentRows() {
+    AccountStatement.openCommentModal({ rows : vm.selectedRows })
+    .then(function (res) {
+      if (!res) { return; }
+      Notify.success('ACCOUNT_STATEMENT.SUCCESSFULLY_COMMENTED');
+      load(AccountStatement.filters.formatHTTP(true));
```

Comment:
Is it necessary to reload all of the data in the account statement just because comments are added? I propose that on confirmation from the server we update the data locally so that it reflects these changes. 

With editing financial information I think it is important to ensure the client and the server have the same data, however with comments I think an optimistic local update should work well. 

What do you think?